src/Language/PureScript/PSString.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Language.PureScript.PSString (PSString, mkString, renderJSON, toUTF16CodeUnits, codePoints) where
+
+import Prelude.Compat
+import Numeric (showHex)
+import Data.List (unfoldr)
+import Data.Monoid ((<>))
+import Data.String (IsString(..))
+import Data.Text (Text)
+import Data.Word (Word16)
+import qualified Data.Aeson as A
+import qualified Data.Text as T
+
+-- |
+-- Strings in PureScript are sequences of UTF-16 code units, which do not
+-- necessarily represent UTF-16 encoded text. For example, it is permissible
+-- for a string to contain *lone surrogates,* i.e. characters in the range
+-- U+D800 to U+DFFF which do not appear as a part of a surrogate pair.
+--
+--
+newtype PSString = PSString [Word16]
+  deriving (Eq, Ord, A.FromJSON, Monoid)
+
+instance Show PSString where
+  show = show . codePoints
```

Comment:
I would love to do that. Incorrect assumptions about `show` are the source of *so many* bugs.